### ISSUE TITLE
train_tensors in BaseDataset is mis-interpreted

### ISSUE THREAD

**Issue by nabenabe0928** [automl/Auto-PyTorch#356]
The continuation of [issue#352](https://github.com/automl/Auto-PyTorch/issues/352).

if I understand the code correctly, we are assuming that `train_tensors[0]` is always feature tensors and `train_tensors[1]` is always labels.
However, `Dataset` in `torch` assumes the following structure:
```python
In [1]: import torchvision.datasets

In [2]: train_tensors = torchvision.datasets.CIFAR10('cifar10/')

In [3]: train_tensors[0]
Out[3]: (<PIL.Image.Image image mode=RGB size=32x32 at 0x7F8B6DAD3370>, 6)

In [4]: train_tensors[1]
Out[4]: (<PIL.Image.Image image mode=RGB size=32x32 at 0x7F8B6DAD33A0>, 9)
```

It indicates that the `__getitem__` of `Dataset` (actually it applies to  other `Dataset` child classes as well) returns:
```
train_tensors[i] := the i-th instance in the given dataset
```

It does not matter unless we use torch stuff, but those stuffs will cause many issues when we try to merge image tasks

**Comment by ravinkohli** [comment]
Hey, actually we do follow the structure of a `Dataset` defined in `torch` see [here](https://github.com/automl/Auto-PyTorch/blob/06e67de5017b4cccad9398e24a3d9f0bd8176da3/autoPyTorch/datasets/base_dataset.py#L205). Specifically for image dataset, for `FilePathDataset` also it follows the structure [here](https://github.com/automl/Auto-PyTorch/blob/06e67de5017b4cccad9398e24a3d9f0bd8176da3/autoPyTorch/datasets/image_dataset.py#L168). I don't think we will have issues when we try to merge image tasks. Moreover, if we were to have issues, we would have had them also for `tabular` tasks as we use a torch `DataLoader` and it works fine.
